src/main/java/donnafin/logic/commands/AppendCommand.java

```diff
@@ -54,12 +72,26 @@
      * @param policy the new policy being added.
      */
     public AppendCommand(PersonAdapter personAdapter, Policy policy) {
+        requireNonNull(personAdapter);
+        requireNonNull(policy);
         this.personAdapter = personAdapter;
         this.hashableNewValue = policy;
         this.editor = pa -> {
             Set<Policy> policies = new HashSet<>(pa.getSubject().getPolicies());
-            policies.add(policy);
-            pa.editPolicies(policies);
+            boolean containsSimilarName = policies.stream().anyMatch(item ->
+                    item.getName().equalsIgnoreCase(policy.getName()));
```

Comment:
tbh why don't you use `.anyMatch(item -> item.isWeakDuplicate(policy))`. Same could be used for assets and liabilities. If not tbh, you can remove the `isWeakDuplicate` function from all the **attribute** classes.

Reply:
Feel free to update the `isWeakDuplicate` methods for policies, assets and liabilities. You may also wanna do a directory wide search for '@Donny' btw.

Reply:
Do you mean that there is already a ```isWeakDuplicate``` function or do you want me to make one?

Reply:
Oops I guess the function was named isPossibleDuplicate. Rip should that be renamed as isWeakDuplicates?